tests/testcases.docopt

```diff
@@ -950,8 +950,35 @@ local options: --baz
 other options:
```

Comment:
As you say in one of the commit messages:

> Note that docopt 0.6.2 recognises option descriptions in the text
> prior to the usage section, but this change does not, as it seems like
> an unintended side-effect of the previous parser's implementation, and
> seems unlikely to be used in practice.

How about we add a test for this:

```
# lines before the usage section that start with --
# are not treated as options.
r"""My CLI program

--speed is ignored as an option

usage: prog --direction
"""
# Note that "--speed" is not present
$ prog
{"--direction": false}
$ prog --direction
{"--direction": true}
```

Reply:
Actually, docopt.org says "Every line that starts with "-" or "--" (not counting spaces) is treated as an option description, e.g.:" so I think we want to stick with that. People are going to be reading that spec, and it's so stable I don't want to mess with it.

That does really throw a wrench in this implementation. What do you think, re-work this implementation to allow for --options to be scattered anywhere?

Reply:
It's not a problem to allow that. We just need to include `sections.before_usage` in the string that gets passed to `parse_options()`. I think you're right, better to stick to the 0.6.2 behaviour.

Reply:
- [ ] Include content in `before_usage` when parsing option descriptions